### ISSUE TITLE
20CVPR # Semi-Supervised Semantic Segmentation with Cross-Consistency Training

### ISSUE THREAD

**Issue by XFeiF** [XFeiF/Video_PaperNotes#33]
[pdf](https://arxiv.org/pdf/2003.09005.pdf)  
[Code-PyTorch](https://github.com/yassouali/CCT)  

**Authors:**
Ouali, Yassine and Hudelot, Celine and Tami, Myriam.   

![](https://github.com/XFeiF/Video_PaperNotes/blob/master/imgs/CCT.png?raw=true)

**Comment by XFeiF** [comment]
**Motivation:**  
**The cluster assumption**: the decision boundary should lie in low density regions. While the authors observe that the cluster assumption is violated at the input level, given that the low density regions do not align with the class boundaries. On the contrary, for the encoder's outputs, the cluster assumption is maintained where the class boundaries have high average distance, thus corresponding to low density regions.  
In one word, in semantic segmentation: **the low density regions are more apparent within the hidden representations than within the inputs.**  The estimation method of the density is shown in the picture below.  
![](https://github.com/XFeiF/Video_PaperNotes/blob/master/imgs/CCT_assumption.png?raw=true)

**Main idea:**   
The authors propose cross-consistency training, where an invariance of the predictions is enforced over different perturbations applied to the outputs of the encoder. Concretely, a shared encoder and a main decoder are trained in a supervised manner using the available labeled examples. To leverage the unlabeled examples, they enforce consistency between the main decoder predictions and those of the auxiliary decoders, taking as inputs different perturbed versions of the encoder’s output, and consequently, improving the encoder’s representations.

**The key to the idea:**   
The auxiliary decoders take as inputs different perturbed versions of the encoder’s output. The **perturbing** methods are very important! And then they enforce consistency between the main decoder predictions and those of the auxiliary decoders.  
- Feature-based perturbations. They consist of either injecting noise into or dropping some of the activations of encoder’s output feature map **z**.
- Prediction-based perturbations: They consist of adding perturbations based on the main decoder’s prediction yˆ = g(z) or that of the auxiliary decoders. We consider masking based perturbations (Con-Msk, Obj-Msk and G-Cutout) in addition to adversarial perturbations (I-VAT).  
- Random perturbations: Spatial dropout.
